### ISSUE TITLE
Add the ability to use private resolvers

### ISSUE THREAD

**Issue by jedisct1** [alterstep/dnscrypt-osxclient#61]
A major limitation in your client is that we can only use public resolvers from the CSV file.

Before the autoupdate mechanism, we could just change the CSV file to add private resolvers. Now, it's not really an option any more.

How can your app be used with private resolvers?

**Comment by Alohajoe5** [comment]
Please, yes, the ability to add/edit the csv file was very helpful.

**Comment by hd-deman** [comment]
It's became more actual since we have [dnscrypt server docker container] (https://github.com/jedisct1/dnscrypt-server-docker)

**Comment by informafix** [comment]
Yes please, if you have time to add it, this option is a must have.

**Comment by mk2soldier** [comment]
+1

**Comment by xxdesmus** [comment]
+1 very much needed.

**Comment by xxdesmus** [comment]
Figured out a solution around this.

Just edit the local CSV file to add your our private resolver (or whatever resolver your want to add) -->

```
sudo vim /Library/PreferencePanes/DNSCrypt.prefPane/Contents/Resources/usr/share/dnscrypt-proxy/dnscrypt-resolvers.csv
```

(note the sudo there -- you'll need root temporarily to write to the file)

then just add the new row you need. Make sure you follow the format similar to [https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-resolvers.csv](https://github.com/jedisct1/dnscrypt-proxy/blob/master/dnscrypt-resolvers.csv)

that's it, open the preference pane again and select the resolver you just addd.

**Comment by mysqlbranch** [comment]
+1 Good idea.

**Comment by soakes** [comment]
@xxdesmus thats a great idea, however everyone should beware that when the client updates/refreshes, it will replace the CSV file which was happening to me.

The solution i've found is, blocking https access to download.dnscrypt.org which seems to be where the updated CSV's are being pulled from.  Thankfully the updates to the program seem to be coming from updates.dnscrypt.org so it doesn't effect the list updates.

If you use little snitch 3, you can do a simple rule as shown in the screenshot below, failing that you could adjust your hosts file and point download.dnscrypt.org to localhost which should serve the same purpose. 

https://www.dropbox.com/s/c1b66shtezec729/Screenshot%202017-05-25%2017.09.25.png?dl=0

I do suggest tho that the client gets adjusted so that you can fead it a list of private servers rather then everyone having to add a few hacks just to use your own recursion servers.
